### ISSUE TITLE
Any plans to make klayjs-d3 compatible with d3-v4?

### ISSUE THREAD

**Issue by rcoaguila** [OpenKieler/klayjs-d3#13]
Hi, I would like to use this library but my project uses d3-v4. I saw in issue #11 that the version was fixed to ^3.5.0, are there any plans to make it compatible with v4? Or can you provide me some pointers to make this change?

Thanks!

**Comment by uruuru** [comment]
Yes. Although I don't know when I find time for it. Currently the klayjs-d3 script assumes that the `d3` object exists and [calls](https://github.com/OpenKieler/klayjs-d3/blob/master/src/js/klay-d3.js#L371) its `rebind` method. With v4 one has to properly import the `d3` object I guess.

**Comment by AnthonyDry** [comment]
I where able quite easily to use klay-3d with v4 the only big changes is that on the dispatch events you need to us call instead of just using the finish event.  as this dispatch.call('finish', foo). I have a working fork over here. https://github.com/AnthonyDry/klayjs-d3/tree/fixesForD3V4 

Im terrible att pull-request and Im not sure how the ww should work, so it might be better if someone else do a proper rewrite later.

Best regards!

**Comment by chandusekhar** [comment]
Is there any place to find klayjs source code to fix it. Complete javascript not build one.

**Comment by uruuru** [comment]
klayjs is based on Java code, not Javascript. See [#16](https://github.com/OpenKieler/klayjs/issues/16) for the Java source and its [documentation ](https://github.com/OpenKieler/klayjs#elk---eclipse-layout-kernel) for reasons why we do not provide new builds at the moment.
